lib/chess/core/pieces/base_piece.rb

```diff
@@ -0,0 +1,111 @@
+require_relative '../chess'
+require_relative '../../functional/operations'
+require_relative '../../functional/to_string'
+
+module Chess
+  module Core
+    module Pieces
+      class BasePiece
+        include Operation
+        include ToString
+
+        attr_reader :symbol, :team, :current_cell
+
+        Vector = Struct.new(:enabled, :deltas)
+
+        def initialize(symbol, team, cell, captured = false)
+          @current_cell = cell
+          @cells_history = []
+          @symbol = symbol
+          @captured = captured
+          @team = team
+          @graph = nil
+
+          @current_cell.occupy_with(self)
+        end
+
+        def move_to(target_cell, cells)
+          return unless can_move_to?(target_cell, cells)
+
+          update_current_cell_to(target_cell)
+        end
+
+        def can_move_to?(target_cell, cells)
+          return false if target_cell.team == @team
+          return evaluate_with_one_move(target_cell, cells) if can_move_once_at_tyme?
+
+          evaluate_with_path(target_cell, cells)
+        end
```

Comment:
```suggestion
        def can_move_to?(target_cell, cells)
          return false if target_cell.team == @team
          return evaluate_with_one_move(target_cell, cells) if can_move_once_at_time?

          evaluate_with_path(target_cell, cells)
        end
```

Reply:
done, and moved to each particular piece's class